server/src/main/java/org/opensearch/index/compositeindex/datacube/startree/aggregators/SumValueAggregator.java

```diff
@@ -7,63 +7,69 @@
  */
 package org.opensearch.index.compositeindex.datacube.startree.aggregators;
 
-import org.apache.lucene.util.NumericUtils;
-import org.opensearch.index.compositeindex.datacube.MetricStat;
 import org.opensearch.index.compositeindex.datacube.startree.aggregators.numerictype.StarTreeNumericType;
 import org.opensearch.search.aggregations.metrics.CompensatedSum;
 
 /**
  * Sum value aggregator for star tree
  *
+ * <p>This implementation follows the Kahan summation algorithm to improve the accuracy
+ * of the sum by tracking and compensating for the accumulated error in each iteration.
+ *
+ * @see <a href="http://en.wikipedia.org/wiki/Kahan_summation_algorithm">Kahan Summation Algorithm</a>
+ *
  * @opensearch.experimental
  */
 public class SumValueAggregator implements ValueAggregator<Double> {
 
-    public static final StarTreeNumericType VALUE_AGGREGATOR_TYPE = StarTreeNumericType.DOUBLE;
+    private final StarTreeNumericType starTreeNumericType;
+
     private double sum = 0;
     private double compensation = 0;
     private CompensatedSum kahanSummation = new CompensatedSum(0, 0);
 
-    private StarTreeNumericType starTreeNumericType;
-
     public SumValueAggregator(StarTreeNumericType starTreeNumericType) {
         this.starTreeNumericType = starTreeNumericType;
     }
 
-    @Override
-    public MetricStat getAggregationType() {
-        return MetricStat.SUM;
-    }
-
-    @Override
-    public StarTreeNumericType getAggregatedValueType() {
-        return VALUE_AGGREGATOR_TYPE;
-    }
-
     @Override
     public Double getInitialAggregatedValueForSegmentDocValue(Long segmentDocValue) {
         kahanSummation.reset(0, 0);
-        kahanSummation.add(starTreeNumericType.getDoubleValue(segmentDocValue));
+        if (segmentDocValue != null) {
+            kahanSummation.add(starTreeNumericType.getDoubleValue(segmentDocValue));
+        } else {
+            kahanSummation.add(getIdentityMetricValue());
+        }
         compensation = kahanSummation.delta();
         sum = kahanSummation.value();
         return kahanSummation.value();
     }
 
+    // we have overridden this method because the reset with sum and compensation helps us keep
+    // track of precision and avoids a potential loss in accuracy of sums.
     @Override
     public Double mergeAggregatedValueAndSegmentValue(Double value, Long segmentDocValue) {
-        assert kahanSummation.value() == value;
+        assert value == null || kahanSummation.value() == value;
         kahanSummation.reset(sum, compensation);
```

Comment:
Can we remove this ? except for 0, 0 reset - we don't need any other resets right ?